src/main/java/org/elasticsearch/action/bulk/TransportShardBulkAction.java

```diff
@@ -130,10 +130,9 @@ protected ShardIterator shards(ClusterState clusterState, InternalRequest reques
     }
 
     @Override
-    protected Tuple<BulkShardResponse, BulkShardRequest> shardOperationOnPrimary(ClusterState clusterState, PrimaryOperationRequest shardRequest) {
+    protected Tuple<BulkShardResponse, BulkShardRequest> shardOperationOnPrimary(ClusterState clusterState, PrimaryOperationRequest shardRequest, IndexShard indexShard) {
```

Comment:
I wonder if we should also pass the IndexService since we already have it in the base class. Saves on the lookup.